contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -1954,40 +1977,43 @@ class tsqlBuilder : public tsqlCommonMutator
 
 	void exitChar_string(TSqlParser::Char_stringContext *ctx) override
 	{
-		if (ctx->STRING())
-		{
-			std::string str = ctx->STRING()->getSymbol()->getText();
+		std::string str = getFullText(ctx);			
 
-			if (str.front() == 'N')
-			{
-				// This is only to make the assertion on str.front() easy (below)
-				str.erase(0, 1);	
-			}
+		if (str.front() == 'N')
+		{
+			// This is only to make the assertion on str.front() easy (below)
+			str.erase(0, 1);	
+		}
 
-			if (str.front() == '"')
-			{	
-				Assert(str.back() == '"');
-				
-				// Change double-quoted string to single-quoted string:
-				str = rewriteDoubleQuotedString(str);	
-				size_t startPosition = ctx->start->getStartIndex();
-				rewritten_query_fragment.emplace(std::make_pair(startPosition, std::make_pair(::getFullText(ctx), str)));
-			}
-			else
-			{
-				// This is a single-quoted string, no further action needed
-				Assert(str.front() == '\'');
-				Assert(str.back() == '\'');		    						
-			}
+		if (str.front() == '"')
+		{	
+			Assert(str.back() == '"');
+			
+			// Change double-quoted string to single-quoted string:
+			str = rewriteDoubleQuotedString(str);	
+			size_t startPosition = ctx->start->getStartIndex();
+			if (in_execute_body_batch_parameter) startPosition += fragment_EXEC_prefix.length(); // add length of prefix prepended internally for execute_body_batch  
+			rewritten_query_fragment.emplace(std::make_pair(startPosition, std::make_pair(::getFullText(ctx), str)));
+		}
+		else
+		{
+			// This is a single-quoted string, no further action needed
+			Assert(str.front() == '\'');
+			Assert(str.back() == '\'');		    						
```

Comment:
We should move this part to enterChar_string() too.

Reply:
This is more of a sanity check but actually it does not add any value given the grammar rules.